viite-UI/src/view/NodeLayer.js

```diff
@@ -69,6 +69,141 @@
         nodePointTemplateLayer.setOpacity(opacity);
         junctionTemplateLayer.setOpacity(opacity);
       };
+      /**
+       * Type of interactions we want the map to be able to respond.
+       * A selected feature is moved to a new/temporary layer out of the default roadLayer.
+       * This interaction is restricted to a single click (there is a 250 ms enforced
+       * delay between single clicks in order to differentiate from double click).
+       * @type {ol.interaction.Select}
+       */
+      var junctionPointTemplateClick = new ol.interaction.Select({
+        //Multi is the one en charge of defining if we select just the feature we clicked or all the overlapping
+        multi: false,
+        //This will limit the interaction to the specific layer, in this case the layer where the roadAddressLinks are drawn
+        layers: [junctionTemplateLayer],
+        //Limit this interaction to the singleClick
+        condition: ol.events.condition.singleClick
+      });
+      junctionPointTemplateClick.set('name','junctionPointTemplateClickInteractionNL');
+
+      // var junctionPointClick = new ol.interaction.Select({
+      //   //Multi is the one en charge of defining if we select just the feature we clicked or all the overlapping
+      //   multi: false,
+      //   //This will limit the interaction to the specific layer, in this case the layer where the roadAddressLinks are drawn
+      //   layers: [junctionMarkerLayer],
+      //   //Limit this interaction to the singleClick
+      //   condition: ol.events.condition.singleClick
+      // });
+      // junctionPointClick.set('name','junctionPointClickInteractionNL');
+
+
+
+      /**
+       * We now declare what kind of custom actions we want when the interaction happens.
+       * Note that 'select' is triggered when a feature is either selected or deselected.
+       * The event holds the selected features in the events.selected and the deselected in event.deselected.
+       *
+       * In this particular case we are fetching every node point template marker in view and
+       * sending them to the selectedNode.open for further processing.
+       */
+
+      junctionPointTemplateClick.on('select', function (event) {
+        // var selected = _.find(event.selected, function (selectionTarget) {
+        //   return !_.isUndefined(selectionTarget.nodePointTemplateInfo);
+        // });
+        // sets selected mode by default - in case a node point is clicked without any mode
+        var selected = _.find(event.selected, function (selectionTarget) {
+          return !_.isUndefined(selectionTarget.junctionPointTemplateInfo);
+        });
+        eventbus.trigger('junctionEdit:selected', selected.junctionPointTemplateInfo.junctionId);
+        // if (!_.isUndefined(selected) && applicationModel.selectedToolIs(LinkValues.Tool.Unknown.value)) {
```

Comment:
Check this comments/file to remove unnecessary code/comments.

Reply:
Fixed and pushed to Git.